src/globus_sdk/config/env_vars.py

```diff
@@ -70,12 +61,28 @@ def _load_var(
     return value
 
 
-def _bool_cast(value: t.Any, default: t.Any) -> bool:  # pylint: disable=unused-argument
+def _ssl_verify_cast(
+    value: t.Any, default: t.Any  # pylint: disable=unused-argument
+) -> bool | str:
     if isinstance(value, bool):
         return value
-    elif not isinstance(value, str):
-        raise ValueError(f"cannot cast value {value} of type {type(value)} to bool")
-    return _str2bool(value)
+    if not isinstance(value, (str, pathlib.Path)):
+        msg = f"Value {value} of type {type(value)} cannot be used for SSL verification"
+        raise ValueError(msg)
+    if isinstance(value, str):
+        if value.lower() in {"y", "yes", "t", "true", "on", "1"}:
+            return True
+        if value.lower() in {"n", "no", "f", "false", "off", "0"}:
+            return False
+        if os.path.isfile(value):
+            return value
+    if isinstance(value, pathlib.Path):
+        if value.is_file():
+            return str(value.absolute())
+    raise ValueError(
+        f"SSL verification {value} is not a valid boolean value "
+        "nor a path to a file that exists"
```

Comment:
Nit: not -> or, or neither -> nor; but not -> nor looks awkward to me. (I'm not certain about this being a correctness thing, but it reads in an unusual way.)

```suggestion
        "or a path to a file that exists"
```

Reply:
"Not" and "nor" are correct, but I can rephrase to something like

> SSL verification value must be a boolean value or a path to a file that exists

Reply:
New rev coming